From 88a1cd85cbc73631d67406b3ad740ee7bd36bc8b Mon Sep 17 00:00:00 2001
Patch-Source: https://github.com/dotnet/coreclr/pull/28227
From: Antoine Martin <dev@ayakael.net>
Date: Sat, 4 Dec 2021 05:03:48 +0000
Subject: [PATCH 1/1] Fix non-portable distrorid alpine generation

<dotnet/coreclr>/init-distro-id.sh generates incorrect DistroRid on Alpine.
While the expected DistroRid (alpine-x.xx-xxx) should only include macro
version, init-distro-id.sh includes the micro version. This patches it
to cut off the trailing subversion off of VERSION_ID by treating it 
the same way RHEL's VERSION_ID is treated.

---
 init-distro-rid.sh | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/init-distro-rid.sh b/init-distro-rid.sh
index 49d8981c4f..7cec3cf649 100755
--- a/init-distro-rid.sh
+++ b/init-distro-rid.sh
@@ -51,7 +51,7 @@ initNonPortableDistroRid()
             # We have forced __PortableBuild=0. This is because -portablebuld
             # has been passed as false.
             if (( ${isPortable} == 0 )); then
-                if [ "${ID}" == "rhel" ]; then
+                if [ "${ID}" == "rhel" ] || [ "${ID}" = "alpine" ]; then
                     # remove the last version digit	
                     VERSION_ID=${VERSION_ID%.*}
                 fi
-- 
2.34.0

